airflow_dags/dag.py

```diff
@@ -140,3 +143,58 @@ def get_args_helper(loaders: List[DoltTableLoader], message: str, remote_url: st
 raw_neural_code_search_eval = BashOperator(task_id='import-data',
                                            bash_command='{{conf.get("core", "dags_folder")}}/neural_code_search_eval/import_from_source.pl ',
                                            dag=neural_code_search_eval_dag)
+
+# Wikipedia word frequency
+WIKIPEDIA_REPO = 'Liquidata/wikipedia-word-frequency'
+DUMP_DATE = datetime.now() - timedelta(days=4)
+FORMATTED_DATE = DUMP_DATE.strftime("%-m-%d-%y")
+
+# XML dumps released on the 1st and 20th of every month. This job should run 4 days after.
+wikipedia_dag = DAG(
+    'wikipedia-word-frequency',
+    default_args=get_default_args_helper(datetime(2019, 10, 18)),
+    schedule_interval='0 8 5,24 * *')
+
+
+def define_wikipedia_dag():
+    wikipedia_no_filter = PythonOperator(
+        task_id='no_filter',
+        python_callable=dolthub_loader,
+        op_kwargs=get_args_helper(get_wikipedia_loaders('none'),
+                                  'Update Wikipedia word frequencies for {} XML dump'.format(FORMATTED_DATE),
+                                  WIKIPEDIA_REPO),
+        dag=wikipedia_dag)
+
+    wikipedia_base_branch = PythonOperator(
+        task_id='base_branch',
+        python_callable=wiki_create_and_push_branch,
+        op_kwargs=(dict(remote=WIKIPEDIA_REPO,
+                        branch_name=FORMATTED_DATE)),
+        dag=wikipedia_dag)
+
+    wikipedia_base_branch.set_upstream(wikipedia_no_filter)
+
+    filters = ['no_numbers', 'no_abbreviations', 'ASCII_only', 'strict', 'convert_to_ASCII', 'stemmed']
+    for word_filter in filters:
+        wikipedia_with_filter = PythonOperator(
+            task_id=word_filter,
+            python_callable=dolthub_loader,
+            op_kwargs=get_args_helper(get_wikipedia_loaders(word_filter),
+                                      'Update Wikipedia word frequencies with {} filter for {} XML dump'.format(
+                                          word_filter, FORMATTED_DATE),
+                                      WIKIPEDIA_REPO,
+                                      branch='{}/filter_{}'.format(FORMATTED_DATE, word_filter)),
+            dag=wikipedia_dag)
+
+        wikipedia_with_filter.set_upstream(wikipedia_base_branch)
+
+    wikipedia_cleanup = PythonOperator(
+        task_id='cleanup',
+        python_callable=wiki_cleanup,
+        dag=wikipedia_dag)
+
+    wikipedia_cleanup.set_upstream(wikipedia_with_filter)
```

Comment:
@oscarbatori should each filter branch have it's own name? I don't want `wikipedia_cleanup` to run until after each branch has been created and pushed

Reply:
This would require a bit of research from me. All my DAGs have been simple. If you fire up airflow webserver, it will draw you a picture of your DAG. What does that look like?

Reply:
I think you want to define `wikipedia_cleanup` before the loop, and then set the dependency by calling `set_downstream` inside the loop. See [this](https://stackoverflow.com/questions/43053354/airflow-task-to-refer-to-multiple-previous-tasks) SO post for details and an example.

Reply:
@timsehn `stemmed` is the last filter in the list so this looks right
![Screen Shot 2019-10-30 at 2 25 55 PM](https://user-images.githubusercontent.com/29443194/67900210-3a2c8700-fb21-11e9-9cc9-190ecff1feda.png)

Reply:
As we yelled at each other ion the office, all those jobs need to point into cleanup. So you need to add each one as an upstream of cleanup.

Reply:
@timsehn @oscarbatori like this?
![Screen Shot 2019-10-30 at 2 41 30 PM](https://user-images.githubusercontent.com/29443194/67901183-66e19e00-fb23-11e9-9234-0333bc11a9f5.png)

Reply:
much better

Reply:
Pushed that change [24e32ad](https://github.com/liquidata-inc/liquidata-etl-jobs/pull/16/files#diff-c808d3e8184c251eb149e2a49652b3c2R177-R193)